src/main/java/crimson/JobRunner.java

```diff
@@ -0,0 +1,114 @@
+        return runCommand(continuousIntegrationJob, targetLocation, "gradle check");
+    }
+
+    /**
+     * Runs the command "gradle build" on a previously cloned repo to build it
+     * @param continuousIntegrationJob the job which triggered the cloning of the repo
+     * @param targetLocation the location of the cloned repo
+     * @return a ContinuousIntegrationJobTaskOutput object containing information about the building process
+     */
+    public ContinuousIntegrationJobTaskOutput runBuild(ContinuousIntegrationJob continuousIntegrationJob,
+                                                       String targetLocation) throws Exception {
+        return runCommand(continuousIntegrationJob, targetLocation, "gradle build");
+    }
+
+    /**
+     * Runs the command "gradle test" on a previously cloned repo to run tests
+     * @param continuousIntegrationJob the job which triggered the cloning of the repo
+     * @param targetLocation the location of the cloned repo
+     * @return a ContinuousIntegrationJobTaskOutput object containing information about the testing process
+     */
+    public ContinuousIntegrationJobTaskOutput runTest(ContinuousIntegrationJob continuousIntegrationJob,
+                                                       String targetLocation) throws Exception {
+        return runCommand(continuousIntegrationJob, targetLocation, "gradle test");
+    }
+
+    /**
+     * Runs an arbitrary command on the command line and logs information about the process
+     * @param continuousIntegrationJob the job which triggered the cloning of the repo
+     * @param targetLocation the path where the command should be executed
+     * @return a ContinuousIntegrationJobTaskOutput object containing information about the process
+     */
+    public ContinuousIntegrationJobTaskOutput runCommand(ContinuousIntegrationJob continuousIntegrationJob,
+                                                         String targetLocation, String command) throws Exception{
+        ContinuousIntegrationJobTaskOutput jobTaskOutput = new ContinuousIntegrationJobTaskOutput();
+        java.io.ByteArrayOutputStream baos = new java.io.ByteArrayOutputStream();
+        java.io.PrintStream ps = new java.io.PrintStream(baos);
+        java.io.PrintStream old = System.out;
+        System.setOut(ps);
+        ProcessBuilder builder = new ProcessBuilder();
+        builder.command(command);
+        builder.directory(new File(targetLocation));
+        Process process = builder.start();
+        int exitCode = process.waitFor();
+        jobTaskOutput.exitCode = exitCode;
+        if (exitCode != 0 ) {
+            jobTaskOutput.ErrorOutput = String.valueOf(process.getErrorStream());
+        }
+        continuousIntegrationJob.succeeded = jobTaskOutput.exitCode == 0;
+        jobTaskOutput.StandardOutput = baos.toString();
+        System.out.flush();
+        System.setOut(old);
+        assert jobTaskOutput.exitCode == 0;
```

Comment:
I think if we assert the exitCode `=0` here, we won't able to notify the pusher that his/her push is failed.

Reply:
@edv1n